connect/transforms/src/test/java/org/apache/kafka/connect/transforms/CastTest.java

```diff
@@ -320,6 +321,40 @@ public void castWholeRecordValueSchemalessUnsupportedType() {
         xformValue.apply(new SourceRecord(null, null, "topic", 0, null, Collections.singletonList("foo")));
     }
 
+    @Test
+    public void castLogicalToPrimitive() {
+        Date day = new Date(MILLIS_PER_DAY);
+        xformValue.configure(Collections.singletonMap(Cast.SPEC_CONFIG,
+            "date:int32,timestamp:int64,time:int64,bigdecimal:int64"));
```

Comment:
It would be great to have a few more test cases to cover more scenarios:
1. cast to types other than `int32` and `int64` (e.g., `float64` and `float32`), including where we lose precision
2. verify we can cast these to strings
3. verify the cast fails in expected ways
4. casting null values should not fail